### DISCUSSION TITLE
Validating settings in pull requests

### DISCUSSION THREAD

**Discussion by martinm82** [github/safe-settings#345] [General]
I know that there is the safe-settings validator but it does not give users the best experience in case they make some mistakes in the configuration file, e.g. typos but as well wrong attribute values (e.g. bool vs. string vs. list, etc.)

Do you think it would be possible to use the GitHub API OpenAPI spec and run it against the settings yaml files and spit out errors in case any issues would be discovered. Or is there an even easier solution for validating the yaml file?

**Comment by svg153** [comment]
I already tested the latest update of [2.1.0-rc.2](https://github.com/github/safe-settings/releases/tag/2.1.0-rc.2), and the reporting was greatly improved both PR and default branch. (Thanks, @decyjphr, nice job!).
- The validation in the PR [now creates a PR comment](https://github.com/github/safe-settings?tab=readme-ov-file#pull-request-workflow) with the addition, deletions, and modifications.
- Now, safe-setting creates a [check in the last commit of the default branch](https://github.com/github/safe-settings?tab=readme-ov-file#error-handling) to report errors or the completion of applying the changes after the merge.
